demo/api/src/products/custom-product.resolver.ts

```diff
@@ -1,18 +1,32 @@
 import { RequiredPermission } from "@comet/cms-api";
 import { InjectRepository } from "@mikro-orm/nestjs";
 import { EntityRepository } from "@mikro-orm/postgresql";
-import { Mutation, Resolver } from "@nestjs/graphql";
+import { Mutation, Query, Resolver } from "@nestjs/graphql";
+import { TranslationService } from "@src/config/translation.service";
 
 import { Product, ProductStatus } from "./entities/product.entity";
 
 @Resolver(() => Product)
 @RequiredPermission(["products"], { skipScopeCheck: true })
 export class CustomProductResolver {
-    constructor(@InjectRepository(Product) private readonly repository: EntityRepository<Product>) {}
+    constructor(
+        @InjectRepository(Product) private readonly repository: EntityRepository<Product>,
+        private readonly translationService: TranslationService,
+    ) {}
 
     @Mutation(() => Boolean)
     async publishAllProducts(): Promise<boolean> {
         await this.repository.nativeUpdate({ status: { $ne: ProductStatus.Published } }, { status: ProductStatus.Published });
         return true;
     }
+
+    @Query(() => String)
+    async myTranslatedString(): Promise<string> {
+        const language = "de"; // TODO get the language from anywhere, e.g. from the request context, scope or database
+
+        return this.translationService.getIntl(language).formatMessage({
```

Comment:
I was thinking about something like `this.intlService.formatMessage(...)` which acts as an NestJS-adapter for Format.JS. But maybe we don't need a service at all?

Reply:
I think it's a good idea to have a service for global stuff like the translation, and not to solve this hidden internal in a module. But if you like I can revert this change :-)

Reply:
@kaufmo what do you think about a service like this?